### PR TITLE
Add basic test with nodes using Redis storage

### PR DESCRIPTION
**Pull request by reinkrul**
If this PR contains a new test, make sure
- [x] it contains a `docker-compose.yml` file using the `nutsfoundation/nuts-node:master` image (for images that should be replaced during automated testing),
- [x] it can be run with a `run-test.sh` script that is called from _all_ appropriate `run-tests.sh` scripts

### CODE REVIEW

File: storage/redis/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      NUTS_CONFIGFILE: /opt/nuts/nuts.yaml
+    ports:
+      - "21323:1323"
+    volumes:
+      - "./node-B/nuts.yaml:/opt/nuts/nuts.yaml:ro"
+      - "../../tls-certs/nodeB-certificate.pem:/opt/nuts/certificate-and-key.pem:ro"
+      - "../../tls-certs/truststore.pem:/opt/nuts/truststore.pem:ro"
+    healthcheck:
+      interval: 1s # Make test run quicker by checking health status more often
+  redis:
```
Comment (gerardsn) [inline storage/redis/docker-compose.yml:27]: is it okay to use the same db for both nodes in the test?

**Review by gerardsn — APPROVED** [review]

File: storage/redis/run-test.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+do
+   curl -s -X POST http://localhost:11323/internal/vdr/v1/did >/dev/null
+   curl -s -X POST http://localhost:21323/internal/vdr/v1/did >/dev/null
+done
+
+echo "------------------------------------"
+echo "Performing assertions..."
+echo "------------------------------------"
+waitForTXCount "NodeA" "http://localhost:11323/status/diagnostics" 41 10
+waitForTXCount "NodeB" "http://localhost:21323/status/diagnostics" 41 10
```
Comment (gerardsn) [inline storage/redis/run-test.sh:39]: aren't the number of transactions now tracked by the node itself? If so, this test only verifies that the redis storage doesn't kill the node...

Comment (reinkrul) [reply storage/redis/run-test.sh:39]: In memory you mean, so maybe if we restart before asserting?

File: storage/redis/docker-compose.yml (hunk 1 above)
Comment (reinkrul) [reply storage/redis/docker-compose.yml:27]: Yeah, they use a different database prefix

File: storage/redis/run-test.sh (hunk 2 above)
Comment (gerardsn) [reply storage/redis/run-test.sh:39]: yes, that would at least retrieve something from storage before asserting

Comment (woutslakhorst) [reply storage/redis/run-test.sh:39]: TX count is read from the DB when requesting diagnostics

**Review by woutslakhorst — APPROVED** [review]

File: storage/redis/run-test.sh (hunk 2 above)
Comment (reinkrul) [reply storage/redis/run-test.sh:39]: I agree that although the assertion in itself is sound, restarting would be an extra assertion, given it's a storage test.

**Review by gerardsn — APPROVED** [review]

**Pull request merged**